sym_api_client_python/clients/datafeed_client_v2.py

```diff
@@ -0,0 +1,57 @@
+from .api_client import APIClient
+import logging
+import json
+
+class DataFeedClientV2(APIClient):
+    def __init__(self, bot_client):
+        self.bot_client = bot_client
+        self.ackid = ""
+
+    def create_datafeed(self):
+        url = '/agent/v5/datafeeds'
+        headers = {'content-type': 'application/json'}
+        response = self.bot_client.execute_rest_call("POST", url, headers=headers)
+
+        datafeed_id = response.get("id")
+        logging.debug('DataFeedClient/create_datafeed() --> {}'.format(datafeed_id))
+        return datafeed_id
+
+
+    def read_datafeed(self, datafeed_id):
```

Comment:
IMHO, `ackId` should be a param of `read_datafeed` method and caching ackID should be performed in `DatafeedEventsService` class.
Let's imagine datafeed_client is a wrapper for datafeed endpoints provided by agent, so `read_datafeed` refers to the read datafeed endpoints and it has ackId as a param.

Reply:
When I was looking over the java SDK, it's not cached in the DF event service but in the client:
**Services > DatafeedEventsServiceV2** `List<DatafeedEvent> events = datafeedClient.readDatafeed(datafeedId, datafeedClient.getAckId())`

Doing so is not capturing the logic in the DF event service at all as you are calling it back from the DF client V2.

Reply:
So I think you can cache it in DF event service class.

Reply:
Fixed by handling the ack Id as a function parameter.